examples/kernel-api.cpp

```diff
@@ -0,0 +1,432 @@
+        RAJA::statement::Tile<0, RAJA::statement::tile_fixed<TILE_DIM>, RAJA::loop_exec,
+
+          //InitList identifies memory within the tuple which needs to be intialized
+          RAJA::statement::InitLocalMem<RAJA::cpu_tile_mem, RAJA::InitList<0>,
+
+          RAJA::statement::For<1, RAJA::loop_exec,
+            RAJA::statement::For<0, RAJA::loop_exec,
+              //The additional types passed into the lambda statement
+              //are used to specify lambda arguments.
+              //Seg<0> - Uses the 0th entry in the Seg tuple as a lambda argument
+              //Seg<1> - Uses the 1st entry in the Seg tuple as a lambda argument
+              //Offset<0> - Uses the offset from the 0th seg tuple as a lambda argument
+              //Offset<1> - Uses the offset from the 1th seg tuple as a lambda argument
+              //Param<0> - Uses the 0th entry in the Param tuple as a lambda argument
+              RAJA::statement::Lambda<0, Seg<0>, Seg<1>, OffSet<0>, OffSet<1>, Param<0> >
+            >
+          >,
+
+          RAJA::statement::For<0, RAJA::loop_exec,
+            RAJA::statement::For<1, RAJA::loop_exec,
+              //If contiguous Lambda arguments are coming from the same tuple,
+              //they may be merged into a *List<>
+              RAJA::statement::Lambda<1, SegList<0, 1>, OffSetList<0, 1>, Param<0> >
+            >
+          >
+
+          >
+        >
+      >
+    >;
+
+  RAJA::kernel_param<SEQ_EXEC_POL_NEW>( RAJA::make_tuple(RAJA::RangeSegment(0, N_c),
+                                                        RAJA::RangeSegment(0, N_r)),
+
+    RAJA::make_tuple(RAJA_Tile),
+
+    [=](int col, int row, int tx, int ty, TILE_MEM &RAJA_Tile) {
+        RAJA_Tile(ty, tx) = Aview(row, col);
+    },
+
+    [=](int col, int row, int tx, int ty, TILE_MEM &RAJA_Tile) {
+      Atview(col, row) = RAJA_Tile(ty, tx);
+
+  });
+
+  checkResult<int>(Atview, N_c, N_r);
+  // printResult<int>(Atview, N_c, N_r);
+
+
+  //--------------------------------------------------------------------------//
+  std::cout << "\n Running RAJA - with kernel API 2 ...\n";
+
+  std::memset(At, 0, N_r * N_c * sizeof(int));
+
+  using SEQ_EXEC_POL_NEW_2 =
+    RAJA::KernelPolicy<
+      RAJA::statement::Tile<1, RAJA::statement::tile_fixed<TILE_DIM>, RAJA::loop_exec,
+        RAJA::statement::Tile<0, RAJA::statement::tile_fixed<TILE_DIM>, RAJA::loop_exec,
+
+          //InitList identifies memory within the tuple which needs to be intialized
+          RAJA::statement::InitLocalMem<RAJA::cpu_tile_mem, RAJA::InitList<0>,
+
+          RAJA::statement::For<1, RAJA::loop_exec,
+            RAJA::statement::For<0, RAJA::loop_exec,
+              //The additional types passed into the lambda statement
+              //are used to specify lambda arguments.
+             RAJA::statement::Lambda<0, LambdaArgs<seg_t, 0, 1>, LambdaArgs<offset_t, 0,1>, LambdaArgs<param_t, 0> >
```

Comment:
@ajkunen @rhornung67 , I have the alternative container working now for lambda arguments. We can change the name of the container, but functionally its doing the job. I was wondering if we liked this option? It does reduce some of the internals.

Reply:
Great.  I think hiding LambdaArgs from the user would be the best.  This allows the internals to be templated, but present the user with a cleaner interface.  All you need to do is define some aliases like:
`template<camp::idx_t ... args> using Segs = LambdaArgs<seg_t, args...>;`

Reply:
@artv3 I agree with @ajkunen about hiding LambdaArgs from the user and using aliases as he described.

Reply:
👍